flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/TestUtils.java

```diff
@@ -414,4 +419,50 @@ public void clear() {
 			backingList.clear();
 		}
 	}
+
+	static class LocalRecoverableWriterForBucketStateMigrationTest extends LocalRecoverableWriter {
```

Comment:
I think we should move this directly to `BucketStateSerializerTest` and rename it to `AlwaysRelativeLocalRecoverableWriter`, and maybe give the prefix as a parameter. Otherwise `"src/test/resources/"` will appear in more than one place and has to be kept in sync.